### PR TITLE
Add a Cache-Control header to the back end response

### LINKED ISSUE #1928: No cache headers in back end responses
**Affected version(s)**

Contao 4.9

**Description**

In Contao 4.4, any back end request also returned the following headers:

```
Cache-Control: max-age=0, must-revalidate, private
Expires: Fri, 10 Jul 2020 12:56:44 GMT
```

(`Expires` will always be the current time). In Contao 4.9 however, this is not the case any more. This in itself is not problematic, but it breaks the back end, if someone happens to have

```
ExpiresByType text/html "access plus 6 hours"
```

in their `.htaccess`, since this causes the back end requests to be cached by the browser for the configured time.

Shouldn't we disallow caching of back end responses in the private cache with the appropriate response headers? Quite a few users in the Contao community already stumbled over this issue.

### PR DESCRIPTION
**Pull request by leofeyer**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #1928
| Docs PR or issue | -

### CODE REVIEW

**Review by bytehead — DISMISSED** [review]

File: core-bundle/src/Resources/contao/classes/BackendTemplate.php [hunk 1]
```diff
@@ -44,6 +46,21 @@ public function parse()
 		return $strBuffer;
 	}
 
+	/**
+	 * Return a response object
+	 *
+	 * @param bool $blnForceCacheHeaders
```
Comment (Toflar) [inline core-bundle/src/Resources/contao/classes/BackendTemplate.php]: ? :)

Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/BackendTemplate.php]: Removed in 2b1d383e9cbf0bddc29705bda455c93c58878ea5. 😊

**Review by Toflar — APPROVED** [review]

File: core-bundle/src/Resources/contao/classes/BackendTemplate.php [hunk 2]
```diff
@@ -44,6 +46,19 @@ public function parse()
...
+	 * Return a response object
+	 *
+	 * @return Response The response object
+	 */
+	public function getResponse()
+	{
+		$response = parent::getResponse();
+		$response->headers->set('Cache-Control', 'no-cache, no-store');
+
+		return $response->setPrivate();
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/classes/BackendTemplate.php:59]: why did you add `setPrivate`? The response should **not** be privately cacheable by the browser?
